backend/src/main/service/CachingService.ts

```diff
@@ -0,0 +1,85 @@
+import { RedisClient } from 'redis';
+import { CachingObject } from '../data/CachingObject';
+import { Promotion } from '../entity/Promotion';
+import { PromotionDTO } from '../validation/PromotionValidation';
+
+// handles the caching for lat/lon values for restaurants with redis
+export class CachingService {
+  private client: RedisClient;
+
+  // sets redis client for class
+  constructor(client: RedisClient) {
+    this.client = client;
+  }
+
+  // formats promotionDTO for caching
+  async getValuesAndCache(promotion: PromotionDTO): Promise<boolean> {
+    const placeId: string = promotion.placeId;
+    const latitude: number = promotion.lat;
+    const longitude: number = promotion.lon;
+    const result = await this.cacheLatLonValues(placeId, latitude, longitude);
+    return result;
+  }
+
+  // caches the lat/lon values for a promotion's restaurant
+  // key is placeId and value is of the form { lat: x, lon: y }
+  // sets the cached value to expire in 30 days (limit imposed by Google Places API)
```

Comment:
Minor comment, could we follow the format for the way comments are structured for `PromotionController` and our other classes?